snf/inject.go

```diff
@@ -254,15 +257,16 @@ func (s *Sender) SendBulk(pkts [][]byte) error {
 		return err
 	}
 
+	s.guardPkts = pkts
 	s.pkts = s.pkts[:0]
 	s.len = s.len[:0]
 	for _, pkt := range pkts {
-		s.pkts = append(s.pkts, pkt...)
+		s.pkts = append(s.pkts, uintptr(unsafe.Pointer(&pkt[0])))
 		s.len = append(s.len, uint32(len(pkt)))
 	}
 
-	return retErr(C.snf_inject_send_bulk(injHandle(s.InjectHandle), s.timeoutMs,
-		s.flags, (*C.uint8_t)(&s.pkts[0]), C.uint32_t(len(s.pkts)), (*C.uint32_t)(&s.len[0])))
+	return retErr(C.snf_inject_send_bulk(injHandle(s.InjectHandle), s.timeoutMs, s.flags,
+		(*unsafe.Pointer)(unsafe.Pointer(&s.pkts[0])), C.uint32_t(len(s.pkts)), (*C.uint32_t)(&s.len[0])))
```

Comment:
you can just pass `unsafe.Pointer(&s.pkts[0])` and handle it as `uintptr_t *pkts` in C code. Looks simpler.

Reply:
You mean `(*C.uintptr_t)(unsafe.Pointer(&s.pkts[0]))`?

Reply:
if you follow the `C.uintptr_t` declaration then it's just `&s.pkts[0]`.

Reply:
Fixed